File: Game of Thrones-I/GameOfThrones-I.py
'''
Created on 25/07/2014

@author: miguel

    Sample Input : 01
    
    aaabbbb
    Sample Output : 01
    
    YES
    Explanation 
    A palindrome permutation of the given string is bbaaabb. 
    
    Sample Input : 02
    
    cdefghmnopqrstuvw
    Sample Output : 02
    
    NO
    Explanation 
    You can verify that the given string has no palindrome permutation. 
    
    Sample Input : 03
    
    cdcdcdcdeeeef
    Sample Output : 03
    
    YES

'''
import sys
if __name__ == '__main__':
    
    code = raw_input()
    codeSet = list(set(code))
    count = [0]*len(codeSet)
    j = 0
    counter = 0
    for i in range(len(codeSet)):
        for j in range(len(code)):
            if codeSet[i] == code[j]:
                count[i] += 1
        if(len(code) % 2 == 0):
            if count[i] % 2 != 0:
                print('NO')
                sys.exit(0)
        else:
            if count[i] % 2 != 0:
                counter += 1
                if counter > 1:
                    print('NO')
                    sys.exit(0)
    print('YES')
    
    
    
        
    

File: Game of Thrones-I/GameOfThrones_I.java
import java.util.HashSet;
import java.util.Scanner;

public class GameOfThrones_I {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		char[] code = sc.nextLine().toCharArray();
		int counter [];
		HashSet<Character> codeSet = new HashSet<Character>();
		for(int i = 0; i < code.length; i++){
			codeSet.add(code[i]);
		}
		Object[] codeS = codeSet.toArray();
		counter = new int[codeSet.size()];
		for(int i = 0; i < codeS.length; i++){
			for(int j = 0; j < code.length; j++){
				if(codeS[i].equals(code[j])){
					counter[i] += 1;
				}
			}
		}
		if(code.length % 2 == 0){
			for(int i: counter){
				if(i % 2 != 0){
					System.out.println("NO");
					System.exit(0);
				}
			}
		} else {
			int count = 0;
			for(int i : counter){
				if(i % 2 != 0){
					count ++;
					if(count > 1){
						System.out.println("NO");
						System.exit(0);
					}
				}
			}
		}
		System.out.println("YES");
		sc.close();
	}

}
